lib/ansible/parsing/vault/__init__.py

```diff
@@ -349,17 +349,19 @@ def script_is_client(filename):
 
 
 def get_file_vault_secret(filename=None, vault_id=None, encoding=None, loader=None):
-    this_path = os.path.realpath(os.path.expanduser(filename))
 
+    # while files from --vault-password-file are unfracked, other sources are not
```

Comment:
Could you extend this comment to explain more of the motivation and not only the effect?